test/src/test_pf.cpp

```diff
@@ -235,6 +223,40 @@ TEST(Pf, ResampleFirstParticle)
   }
 }
 
+TEST(Pf, ResampleFirstAndLastParticle)
+{
+  const float small_prob = 1.0e-06f;
+  {
+    const std::vector<float> probs =
+        {
+            small_prob, 0.2f, 0.2f, 0.2f, 0.4f - small_prob
+        };
+    const std::vector<float> states =
+        {
+            0.0f, 1.0f, 2.0f, 3.0f, 4.0f
+        };
+    const std::vector<float> expected_resampled_states =
+        {
+            1.0f, 2.0f, 3.0f, 4.0f, 4.0f
+        };
+    testResample(probs, states, expected_resampled_states);
```

Comment:
Please add scoped trace.
https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#adding-traces-to-assertions

Reply:
Done